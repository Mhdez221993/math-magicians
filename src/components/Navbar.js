import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Math Magicians',
    },
    {
      id: 2,
      path: '/',
      text: 'Home',
    },
    {
      id: 3,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 4,
      path: '/quotes',
      text: 'Quotes',
    },
  ];

  return (
    <nav className="navBar">
      <ul className="nav-list">
        {links.map(link => (
          <li key={link.id} className="nav-links">
            <NavLink to={link.path} activeClassName="active-link" exact>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
