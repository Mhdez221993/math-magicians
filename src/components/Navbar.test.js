import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

it('Navbar snapshot test', () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
