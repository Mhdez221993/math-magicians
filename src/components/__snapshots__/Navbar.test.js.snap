// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot test 1`] = `
<nav
  className="navBar"
>
  <ul
    className="nav-list"
  >
    <li
      className="nav-links"
    >
      <a
        aria-current="page"
        className="active-link"
        href="#/"
        onClick={[Function]}
        style={Object {}}
      >
        Math Magicians
      </a>
    </li>
    <li
      className="nav-links"
    >
      <a
        aria-current="page"
        className="active-link"
        href="#/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li
      className="nav-links"
    >
      <a
        aria-current={null}
        href="#/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-links"
    >
      <a
        aria-current={null}
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
